presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java

```diff
@@ -406,6 +412,26 @@ private Object toPrestoValue(Object tpchValue, Type columnType)
         throw new IllegalArgumentException("unsupported column type " + columnType);
     }
 
+    @Override
+    public ConnectorAnalyzeTableMetadata getAnalyzeTableMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, Map<String, Object> properties)
+    {
+        ConnectorTableMetadata tableMetadata = getTableMetadata(session, tableHandle);
+        TableStatisticsMetadata statisticsMetadata = new TableStatisticsMetadata(ImmutableSet.of(), ImmutableSet.of(ROW_COUNT), ImmutableList.of());
+        return new ConnectorAnalyzeTableMetadata(statisticsMetadata, tableMetadata, tableHandle);
```

Comment:
I don't like the fact that tpch supports ANALYZE statement. We ship tpch connector as part of Presto tarball, and users may end up running ANALYZE on these tables. ("since it works, it must be doing something")

I think we shouldn't be doing that. If we do want to do this, we can define connector property `tpch.allow-analyze-mock` and support ANALYZE only when it's set.